File: openapi-generator/php-ze-ph/src/App/DTO/VehicleMaintenanceJ1939CheckEngineLight.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 * J1939 check engine lights.
 */
class VehicleMaintenanceJ1939CheckEngineLight
{
    /**
     * @DTA\Data(field="protectIsOn", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"bool"})
     * @var bool
     */
    public $protect_is_on;
    /**
     * @DTA\Data(field="stopIsOn", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"bool"})
     * @var bool
     */
    public $stop_is_on;
    /**
     * @DTA\Data(field="warningIsOn", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"bool"})
     * @var bool
     */
    public $warning_is_on;
    /**
     * @DTA\Data(field="emissionsIsOn", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"bool"})
     * @var bool
     */
    public $emissions_is_on;
}

File: openapi-generator/clojure/src/samsara_api/specs/vehicle_maintenance_j1939_check_engine_light.clj
(ns samsara-api.specs.vehicle-maintenance-j1939-check-engine-light
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def vehicle-maintenance-j1939-check-engine-light-data
  {
   (ds/opt :protectIsOn) boolean?
   (ds/opt :stopIsOn) boolean?
   (ds/opt :warningIsOn) boolean?
   (ds/opt :emissionsIsOn) boolean?
   })

(def vehicle-maintenance-j1939-check-engine-light-spec
  (ds/spec
    {:name ::vehicle-maintenance-j1939-check-engine-light
     :spec vehicle-maintenance-j1939-check-engine-light-data}))
